include/nw4r/ut/Color.h

```diff
@@ -7,18 +7,31 @@ namespace ut {
 
 struct Color : public _GXColor {
 public:
+    Color() {
+        *this = 0xFFFFFFFF;
+    }
+
+    Color(const Color &rhs) {
+        r = rhs.r;
+        g = rhs.g;
+        b = rhs.b;
+        a = rhs.a;
+    }
+
     Color(u32 val) {
         *this = val;
     }
 
+    ~Color() {}
+
     void operator=(u32 val) {
         ToU32ref() = val;
     }
 
     u32 &ToU32ref() {
         return *reinterpret_cast<u32 *>(this);
     }
-};
+} __attribute__ ((aligned(4)));
```

Comment:
I remember you talking about the alignment on this structure. Are you really sure they did this?
There are surprisingly quite a few NW4R structures that we know have explicit padding members due to DWARF.

Reply:
I'm not sure. Looking through Ketteiban and other games I had on hand, I could not find an instance where `Color` was initialized outside of this alignment. nw4r is hyper-optimized, but EGG does not strike me the same way - I am operating under the assumption that no explicit padding members are in EGG.

Combined with the possibility of e.g. `lwz r0,0xe(r3)` otherwise, this is my preferred solution. I can always adjust it later if necessary.